sktime/forecasting/hf_momentfm_forecaster.py

```diff
@@ -0,0 +1,532 @@
+                " values will be used when fitting.",
+                stacklevel=2,
+            )
+            self._seq_len = 512
+        # in the case the config contains 'forecasting_horizon', we'll set
+        # fh as that, otherwise we override it using the fh param
+        self._fh_config = (
+            self._config["forecasting_horizon"]
+            if "forecasting_horizon" in self._config.keys()
+            else self.fh
+        )
+        if fh is not None:
+            self._fh_input = max(fh.to_relative(self.cutoff))
+        self._fh = self._fh_input if fh is not None else self._fh_config
+        train_val_split = int(len(y) * (1 - self.train_val_split))
+        self._model = momentfm.MOMENTPipeline.from_pretrained(
+            self._pretrained_model_name_or_path,
+            model_kwargs={
+                "task_name": "forecasting",
+                "dropout": self._dropout,
+                "head_dropout": self._head_dropout,
+                "freeze_encoder": self._freeze_encoder,
+                "freeze_embedder": self._freeze_embedder,
+                "seq_len": 512,  # forced to be hard coded
+                "freeze_head": self._freeze_head,
+                "device": self._device,
+                "transformer_backbone": self._transformer_backbone,
+                "forecast_horizon": self._fh,
+            },
+        )
+        self._model.init()
+        self._y_cols = y.columns
+        self._y_shape = y.values.shape
+        train_dataset = momentPytorchDataset(
+            y=y[:train_val_split],
+            fh=self._fh,
+            seq_len=self._seq_len,
+        )
+
+        train_dataloader = DataLoader(
+            train_dataset, batch_size=self.batch_size, shuffle=True
+        )
+
+        val_dataset = momentPytorchDataset(
+            y=y[train_val_split:],
+            fh=self._fh,
+            seq_len=self._seq_len,
+        )
+
+        val_dataloader = DataLoader(
+            val_dataset, batch_size=self.batch_size, shuffle=True
+        )
+
+        criterion = self._criterion
+        optimizer = Adam(self._model.parameters(), lr=1e-4)
+        if self.device == "gpu" and torch.cuda.is_available():
+            self.device = "cuda"
+        else:
+            self.device = "cpu"
+        cur_epoch = 0
+        max_epoch = self.epochs
+
+        # Move the model to the GPU
+        self._model = self._model.to(self.device)
+
+        # Move the loss function to the GPU
+        criterion = criterion.to(self.device)
+
+        # Enable mixed precision training
+        scaler = torch.cuda.amp.GradScaler()
+
+        # Create a OneCycleLR scheduler
+        max_lr = 1e-4
+        total_steps = len(train_dataloader) * max_epoch
+        scheduler = OneCycleLR(
+            optimizer, max_lr=max_lr, total_steps=total_steps, pct_start=0.3
+        )
+
+        # Gradient clipping value
+        max_norm = 5.0
+
+        while cur_epoch < max_epoch:
+            losses = []
```

Comment:
can you extract this into a _run_epoch method. I think this might make everything easier to read, since the methods are smaller